### DISCUSSION TITLE
Is there any way to make an ssr nuxt app refresh every pagelink or is there any way to hide api calls on ssr application just like normal ssr app??

### DISCUSSION THREAD

**Discussion by ZeroCoder9324** [nuxt/nuxt#18865] [Questions]
hello, I'm new in web dev and I have some problems hiding my api calls or endpoint on my ssr nuxt app , I thought by making my site an ssr will hide my api calls from dev tools but I think it's not totally right, because when I go or click any pagelink, it's showing my api calls on that page and somehow I found the cause, it seems that my nuxt app is not reloading/refreshing the page when I go to next page which I think behaving like an spa, where it calls the api on client side. Please help.

**Comment by odranoelBR** [comment]
Hi @ZeroCoder9324 , you are right.

The first page load will hide all your api calls [Isomorphic fetch](https://nuxt.com/docs/getting-started/data-fetching#isomorphic-fetch-and-fetch)

The navigation between pages will show the api requests on the browser network tab. This is the common behaviour of SSR. 
The mains goal of SSR is not about api calls, but about render the entire HTML in backend to give the browser the html ready! Faster loadings, less Js parse etc.

- If you need your pages to full reload between navigation you can't use [Vue router - Router link ](https://router.vuejs.org/api/index.html#routerlink) or the pimped one [Nuxt link](https://nuxt.com/docs/examples/routing/nuxt-link#nuxtlink)
Just an <\a> solve it. (Your pages will work like java JSF pages, PHP pages etc)

- if you concern about the api requests showing in browser is about security, I sugest you [Nuxt security](https://nuxt.com/modules/security) and use a proxy api server from nuxt like [Proxy module](https://nuxt.com/modules/api-party)

I hope have helped and keep styding and coding =)
